### ISSUE TITLE
Make ZPL Orientation configurable

### ISSUE THREAD

**Issue by frazhome** [OpenPrinting/cups#623]
Since the  merge of https://github.com/apple/cups/commit/29500e14c7c977ac56aa520be45092ef8ea766fe the orientation on zebra label printer is rotated by 180 degree.

       /*
        * Rotate 180 degrees so that the top of the label/page is at the
	* leading edge...
	*/

	puts("^POI");

from the ZPL manual:
^POI - Print Orientation Inverted
^PON - Print Orientation normal

some years ago I created the merge request https://github.com/apple/cups/pull/4925

but the merge was canceld.

After some years later I am still fighting against this rotated labels and could not solve all problems.

the idea to change the orientation via -o orientation-requested=6 works mostly, but we are not in a perfect world.

Label-designer and Label-Developer are doing crazy stuff from time to time.

please extend the rastertolabel filter and the drv-file.

This would be very usefull for a lot of user.

regards
Franz

**Comment by michaelrsweet** [comment]
The point of this change was to make the Zebra output consistent with all other label printers and IPP in general.

**Comment by frazhome** [comment]
Hi Michael,

why did you close the isssue with wontfix.

I understand your Idea to keep the output consistent and I like this idea.
But we do not always live in a perfect world and often we have to deal with real world issues.

When this option would we merged a lot of real world problem would be fixed.

When Zebra printers are used in a mixed environement  with raw zpl data stream and cups both datastreams disturbe each other.

When you are dealing with enterprise application like SAP it is not always possible to manipulate the datastream and orientation.

So a configuration options could be very helpfull and when the default is configured with ^POI nobody will have a problem with this and in special cases the administrator is able to fix this kind of orientation issues.

I still strongly hope you will care of this and implement the configuration option!

best regards
Franz

**Comment by michaelrsweet** [comment]
We’ve been down this road before about why the change was done. As for why this is “wontfix”:

- CUPS drivers are deprecated, and LPrint is the “official” replacement printer application for Zebra label printer which offers a default orientation option you can configure
- The proper way to get the orientation you want is through the orientation-requested option - set the default value or specify the value (6 == reverse-portrait) in the print command.
- if you are sending raw ZPL from an application, it is responsible for setting the orientation.

**Comment by frazhome** [comment]
Hello Michael,
I am back again ...

1.  You deprecate the the CUPS drivers, and LPrint the "official" replacement, but I have never seen such a application. I am still working with CUPS as a central Spoolserver for a lot of server applications like SAP ERP. Did you talk to SAP about LPrint? Is there a usefull documentation about this? Are there real live examples?
2. I set the default orientation via "lpadmin -p <printername> -o orientation-requested-default=6. yes, this is working frequently, but not always, cause of formular developer. And is there a interface to show this settings. I extended the template to display this, but there have to be a standard. To manage this via web this is also not possible.
3. You, did not understand. In my scenario, the ZPL-printer is driven from two printserver. And when you have some experience with such thermotransfer printers, you know this devices are acting different to laserprinter. In a perfect World you are right, but the world is different.

In the end I do not understand why you are so strong against this extension to the ZPL. Modify the drv-file, so ROI is the default and nothing changes. In the cornercases admins are able to handle the issues.
In the end this is only a extension of 10 to 15 lines of code and this would be helpfull for a lot of use cases.
